specutils/spectra/spectrum1d.py

```diff
@@ -88,6 +88,12 @@ def __init__(self, flux=None, spectral_axis=None, wcs=None,
             raise LookupError("No WCS object or spectral axis information has "
                               "been given. Please provide one.")
 
+        # Check to make sure the wavelength length is the same in both
+        if flux is not None and spectral_axis is not None:
+            if not spectral_axis.shape[0] == flux.shape[-1]:
+                raise ValueError('Spectral axis ({}) and flux axis ({}) lengths must be the same'.format(
```

Comment:
Maybe this should be 
```suggestion
                raise ValueError('Spectral axis ({}) and the last flux axis ({}) lengths must be the same'.format(
```
?  Because for a vector `Spectrum1D`, "flux axis" is potentially ambiguous